media-source-respec.html

```diff
@@ -1199,11 +1201,14 @@
           indicate that <a def-id="random-access-point"></a> <a def-id="coded-frame"></a> is needed before anything can be added to the
           <a def-id="track-buffer"></a>.</p>
 
-        <p>Each <a def-id="track-buffer"></a> has a <dfn id="track-buffer-ranges">track buffer ranges</dfn> variable that represents the presentation time ranges occupied by the <a def-id="coded-frames"></a>
-          currently stored in the track buffer. For specification purposes, this information is treated as if it were stored in a <a def-id="normalized-timeranges-object"></a>.</p>
+        <p>Each <a def-id="track-buffer"></a> has a <dfn id="track-buffer-ranges">track buffer ranges</dfn> variable that represents the presentation time ranges occupied by the <a def-id="coded-frames"></a> currently stored in the track buffer.
+
+          <p class="note">For track buffer ranges, these presentation time ranges are based on <a def-id="presentation-timestamp">presentation timestamps</a>, frame durations, and potentially coded frame group start times for coded frame groups across track buffers in a muxed <a>SourceBuffer</a>.</p>
 
-        <p class="note">For track buffer ranges, these "presentation time ranges" are based on Presentation Time Stamps (PTS), frame durations, and potentially coded frame group start times for coded frame groups across track buffers in a muxed <a>SourceBuffer</a>.</p>
+          For specification purposes, this information is treated as if it were stored in a <a def-id="normalized-timeranges-object"></a>. Intersected <a def-id="track-buffer-ranges"></a> are used to report <a def-id="hme-buffered"></a>, and must therefore support uninterrupted playback within each range of <a def-id="hme-buffered"></a>.
 
+          <p class="note">These coded frame group start times differ slightly from those mentioned in the <a def-id="coded-frame-processing-algorithm"></a> in that they are the earliest <a def-id="presentation-timestamp"></a> across all track buffers following a discontinuity, which can occur within the <a def-id="coded-frame-processing-algorithm"></a> or result from the <a def-id="coded-frame-removal-algorithm"></a>, regardless of <a def-id="mode"></a>.
+          The threshold for determining disjointness of <a def-id="track-buffer-ranges"></a> is implementation-specific; to reduce unexpected playback stalls, implementations may approximate the <a def-id="coded-frame-processing-algorithm"></a>'s discontinuity detection logic by coalescing adjacent ranges separated by a gap smaller than 2 times the maximum frame duration buffered so far in this <a def-id="track-buffer"></a>. Implementations may also use coded frame group start times as range start times across <a def-id="track-buffer">track buffers</a> in a muxed <a>SourceBuffer</a> to further reduce unexpected playback stalls.</p>
```

Comment:
> [ break at the ";" ? ]
Sounds good to me. I'll do that shortly.

> [ Approximation's derivation? ] 
There is no "maximum frame duration buffered so far" in the coded frame processing algorithm (CFP). However, CFP detects discontinuities when the DTS sequence goes backwards (inapplicable to this approximation derivation) or when:
"If last decode timestamp for track buffer is set and the difference between decode timestamp and last decode timestamp is greater than 2 times last frame duration:"
Note especially the "2 times last frame duration" portion of this. Also note that this approximation text is fully within a non-normative note, and is intended to help implementations and web authors understand that there is lenience (the threshold...is implementation-specific) and that there are some suggestions for reaching better interop here. Would you prefer different, or even normative, text, for this?